### ISSUE TITLE
Problem installing hstaxes

### ISSUE THREAD

**Issue by steveschulze** [spacetelescope/hstaxe#95]
Hi,

I try to install hstaxe on a Mac M1 with Mac OS 13.5.2. I followed the installation instruction on the GitHub home page and the documentation. When I do `pip install hstaxe --no-cache-dir`, it downloads several versions of matplotilb. This throws the following error:

```
Collecting matplotlib (from drizzlepac->hstaxe)
  Downloading matplotlib-3.4.3.tar.gz (37.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.9/37.9 MB 41.4 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.4.2.tar.gz (37.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.3/37.3 MB 40.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.4.1.tar.gz (37.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.3/37.3 MB 40.0 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.4.0.tar.gz (37.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.1/37.1 MB 43.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.3.4.tar.gz (37.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.9/37.9 MB 40.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.3.3.tar.gz (37.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.9/37.9 MB 38.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.3.2.tar.gz (37.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.9/37.9 MB 39.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.3.1.tar.gz (38.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.8/38.8 MB 41.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.3.0.tar.gz (38.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.8/38.8 MB 41.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.2.2.tar.gz (40.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.3/40.3 MB 40.0 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.2.1.tar.gz (40.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.3/40.3 MB 40.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.2.0.tar.gz (40.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.5/40.5 MB 40.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.1.3.tar.gz (40.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.9/40.9 MB 41.4 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.1.2.tar.gz (40.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.9/40.9 MB 41.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.1.1.tar.gz (37.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.8/37.8 MB 41.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.1.0.tar.gz (37.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 37.2/37.2 MB 40.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Downloading matplotlib-3.0.3.tar.gz (36.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 36.6/36.6 MB 34.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [55 lines of output]
      /private/var/folders/fh/s48n0wms0_z12jdfyhml2fw80000gn/T/pip-install-gyp91cbq/matplotlib_9d5bd903e4b34a87bd52c1eba8244b0b/setup.py:33: SetuptoolsDeprecationWarning: The test command is disabled and references to it are deprecated.
      !!
      
              ********************************************************************************
              Please remove any references to `setuptools.command.test` in all supported versions of the affected package.
      
              By 2024-Nov-15, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.
              ********************************************************************************
      
      !!
        from setuptools.command.test import test as TestCommand
      ============================================================================
      Edit setup.cfg to change the build options
      
      BUILDING MATPLOTLIB
                  matplotlib: yes [3.0.3]
                      python: yes [3.10.15 | packaged by conda-forge | (main, Oct
                              16 2024, 01:24:20) [Clang 17.0.6 ]]
                    platform: yes [darwin]
      
      REQUIRED DEPENDENCIES AND EXTENSIONS
                       numpy: yes [not found. pip may install it below.]
            install_requires: yes [handled by setuptools]
                      libagg: yes [pkg-config information for 'libagg' could not
                              be found. Using local copy.]
                    freetype: no  [The C/C++ header for freetype2 (ft2build.h)
                              could not be found.  You may need to install the
                              development package.]
                         png: yes [version 1.6.40]
                       qhull: yes [pkg-config information for 'libqhull' could not
                              be found. Using local copy.]
      
      OPTIONAL SUBPACKAGES
                 sample_data: yes [installing]
                    toolkits: yes [installing]
                       tests: no  [skipping due to configuration]
              toolkits_tests: no  [skipping due to configuration]
      
      OPTIONAL BACKEND EXTENSIONS
                         agg: yes [installing]
                       tkagg: yes [installing; run-time loading from Python Tcl /
                              Tk]
                      macosx: yes [installing, darwin]
                   windowing: no  [Microsoft Windows only]
      
      OPTIONAL PACKAGE DATA
                        dlls: no  [skipping due to configuration]
      
      ============================================================================
                              * The following required packages can not be built:
                              * freetype
                              * Try installing freetype with `brew install
                              * freetype` and pkg-config with `brew install pkg-
                              * config`
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```
To get around this error, I installed Matplotlib before hstaxe, but one of the dependencies in hstaxe still tries to install a particular version of Matplotlib.

Could you help me to install the software?

Cheers,

Steve

**Comment by rosteen** [comment]
Sorry for the difficulty, I'm working on a PR (https://github.com/spacetelescope/hstaxe/pull/98) that will resolve these issues. If you install from that PR for now I believe most of the problems are resolved, I think this command should work to do that:
`pip install git+https://github.com/spacetelescope/hstaxe.git@refs/pull/98/merge`

**Comment by rosteen** [comment]
Closed by #98
